src/_pytest/assertion/rewrite.py

```diff
@@ -937,6 +942,17 @@ def visit_Assert(self, assert_: ast.Assert) -> List[ast.stmt]:
                 ast.copy_location(node, assert_)
         return self.statements
 
+    def visit_NamedExpr(self, name: namedExpr) -> Tuple[namedExpr, str]:
+        # Display the repr of the target name if it's a local variable or
+        # _should_repr_global_name() thinks it's acceptable.
```

Comment:
This was copied from `visit_Name`, perhaps we can update the comment to mention that this handles the walrus operator instead?